FuncGameDev/FSharp/Unity/Update.fs

```diff
@@ -36,14 +42,43 @@ type Updater() =
     member this.checkGameOver() =
         match GameState.instance.gamedata.floor with 
         | 11 ->
-            match highScore with
-            | highScore when highScore |> float > GameState.instance.gamedata.time ->
-                ScoreSavingService.storeScore (GameState.instance.gamedata.time |> string)
-            | _ -> ()
+            match highestRemaining with
+            | highestRemaining when highestRemaining |> float < GameState.instance.gamedata.time ->
+                match lowestTotal with
```

Comment:
I'd suggest unnesting these match statements to reduce code duplication. Right now the lowestTotal code appears twice.